scheduler/test/cook/test/mesos/mesos_mock.clj

```diff
@@ -0,0 +1,844 @@
+(ns cook.test.mesos.mesos-mock
+  (:use clojure.test)
+  (:require [clojure.core.async :as async]
+            [clojure.data.csv :as csv]
+            [clojure.java.io :as io]
+            [clojure.tools.logging :as log]
+            [cook.mesos :as c]
+            [cook.mesos.mesos-mock :as mm]
+            [cook.mesos.share :as share]
+            [cook.test.testutil :refer (restore-fresh-database! create-dummy-group create-dummy-job create-dummy-instance)]
+            [datomic.api :as d]
+            [mesomatic.scheduler :as mesos]
+            [mesomatic.types :as mesos-types])
+  (:import org.apache.curator.framework.CuratorFrameworkFactory
+           org.apache.curator.framework.state.ConnectionStateListener
+           org.apache.curator.retry.BoundedExponentialBackoffRetry))
+
+(defn string->uuid
+  "Parses the uuid if `uuid-str` is a uuid, returns nil otherwise"
+  [uuid-str]
+  (try
+    (java.util.UUID/fromString uuid-str)
+    (catch Exception e
+      nil)))
+
+(defn noop-mock-scheduler
+  []
+  (mesos/scheduler
+    (registered [this driver framework-id master-info])
+    (reregistered [this driver master-info])
+    (offer-rescinded [this driver offer-id])
+    (framework-message [this driver executor-id slave-id data])
+    (disconnected [this driver])
+    (slave-lost [this driver slave-id])
+    (executor-lost [this driver executor-id slave-id status])
+    (error [this driver message])
+    (resource-offers [this driver offers])
+    (status-update [this driver status])))
+
+(deftest make-offer
+  (let [basic-host-info {:hostname "hosta"
```

Comment:
I completely agree with your points. 

I was thinking something more along the lines of:

```clj
(defn host-info
  "Constructs a host-info map with the given data"
  ([hostname slave-id cpus mem ports]
   (host-info hostname slave-id cpus mem ports nil nil nil))
  ([hostname slave-id cpus mem [ports-begin ports-end] cpus-available
    mem-available [ports-available-begin ports-available-end]]
   {:hostname hostname
    :attributes {}
    :resources {:cpus {"*" cpus}
                :mem {"*" mem}
                :ports {"*" [{:begin ports-begin :end ports-end}]}}
    :available-resources (cond-> {:cpus {}, :mem {}, :ports {}}
                                 cpus-available (assoc :cpus {"*" cpus-available})
                                 mem-available (assoc :mem {"*" mem-available})
                                 (and ports-available-begin
                                      ports-available-end) (assoc :ports {"*" [{:begin ports-available-begin
                                                                                :end ports-available-end}]}))
    :slave-id slave-id}))
```

which you could then use in tests, either with available resources or without:

```clj
;; with available resources
(host-info "hosta" "e8ab3892-3023-4546-888d-67cc13eac3cc" 10 100000 [10000 20000] 10 100000 [10000 20000])

;; without
(host-info "hostb" "a5cb3212-3023-4546-888d-67cc13eac3dd" 12 120000 [10000 20000])
```

This avoids the "thing that I expect" function and the host map `def`, while also preventing a bunch of boilerplate and making the tests proper easier to read. What do you think?